cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricStore.java

```diff
@@ -331,17 +331,21 @@ private String getAggregation(MetricDataQuery query) {
 
   @Override
   public void deleteBefore(long timestamp) throws Exception {
-    // Delete all data before the timestamp. null for MeasureName indicates match any MeasureName.
     for (int resolution : resolutions) {
-      // NOTE: we do not purge on TTL the "totals" currently, as there might be system components dependent on it
-      if (TOTALS_RESOLUTION == resolution) {
-        continue;
-      }
-      CubeDeleteQuery query = new CubeDeleteQuery(0, timestamp, resolution, Maps.<String, String>newHashMap());
-      cube.get().delete(query);
+      deleteBefore(timestamp, resolution);
     }
   }
 
+  @Override
+  public void deleteBefore(long timestamp, int resolution) throws Exception {
+    if (TOTALS_RESOLUTION == resolution) {
```

Comment:
This is an awkward method contract that it excludes the `TOTALS_RESOLUTION`. It may be better for this class to understand retention (by reading from cConf) and expose one method `deleteTTLExpired(long currentTimeMillis)`

Reply:
It is a leftover in the MetricStore, will change that